src/components/Pagination/hooks/usePagination.ts

```diff
@@ -0,0 +1,59 @@
+import React, { useMemo } from "react";
+import { BaseUsePaginationProps } from "../type";
+
+export const DOTS = "...";
+
+// WA: Create Array with certain range based on data length
+export const range = (start: number, end: number) => {
+  let length = end - start + 1;
+  return Array.from({ length }, (_, idx) => idx + start);
+};
+
+export const usePagination = (props: BaseUsePaginationProps) => {
+  const { totalPage, pageSize, siblingCount = 1, currentPage } = props;
+  const paginationRange = useMemo(() => {
+    // WA: Pages count is determined as siblingCount + firstPage + lastPage + currentPage + 2*DOTS
+    const totalPageNumbers = siblingCount + 5;
```

Comment:
siblingCount teh apa hib?

sama kenapa ada 2*DOTS di belakang, care to explain this totalPageNumbers variable?

Reply:
- siblingCount itu buat nentuin jumlah page setelah/sebelum atau diantara page tsb.

https://user-images.githubusercontent.com/104057020/178212146-0613aacf-406a-46d1-8a53-533907830f2c.mov

- total page numbers itu jumlah total page yang ditampilin di layar. Kenapa ada 2*DOTS buat handle case ketika kita ada di halaman tengah 

<img width="535" alt="Screen Shot 2022-07-11 at 14 33 11" src="https://user-images.githubusercontent.com/104057020/178212017-9cffb9a7-428c-4e20-a433-6bf08da690da.png">

Reply:
hee i see ok